README.md

```diff
@@ -143,13 +143,16 @@ log.Debug("Available ingredients", "ingredients", ingredients)
 
 ### Options
 
-You can customize the logger with options. Use `log.WithCaller()` to enable
-printing source location. `log.WithTimestamp()` prints the timestamp of each
-log.
+You can customize the logger with options. Use `log.LogOptions{}` to customize
+your new logger.
 
 ```go
-logger := log.New(log.WithTimestamp(), log.WithTimeFormat(time.Kitchen),
-    log.WithCaller(), log.WithPrefix("Baking 🍪 "))
+logger := log.LogOptions{
+    ReportCaller: true,
+    ReportTimestamp: true,
+    TimeFormat: time.Kitchen,
+    Prefix: "Baking 🍪 "
+}.New(os.Stderr)
```

Comment:
this looks a bit unusual to me, I would expect to pass the options to `New`, instead of creating a new logger from the options struct...

the `options.New` semantics reads to me like I'm creating a new log options instance from the lefthand log options 🤔

Reply:
True this might be a bit unusual. How about `New(io.Writer, ...LogOptions)`? 
The example above would be:
```go
logger := log.New(os.Stderr, log.LogOptions{
    ReportCaller: true,
    ReportTimestamp: true,
    TimeFormat: time.Kitchen,
    Prefix: "Baking 🍪 "
})
```

Reply:
Another approach I've seen and quite like is allowing to pass each option:

```go
logger := log.New(log.WithOutput(os.Stderr), log.WithCaller(true), log.WithTimestamp(true), log.WithPrefix("Baking"))
```

This allows the caller of `New` to only customize the options they need to, rather than needing to pass the entire struct and rely on the default values of Go.

Reply:
maybe then `log.From(prefLoger, log.WithPrefix("otherprefix"))`? 🤔

Reply:
another idea would be to do `log.New(os.Stderr, log.LogOptions{ /* ... */ })`

Reply:
ah, just now saw your previous comment suggesting the same thing 🙈

Reply:
> `New(io.Writer, ...LogOptions)`

What I don't love about this one is that that it suggests there's more than one set of options that can be passed. I think what we typically see in Go for cases like this is something more along the lines of:

```go
func New() *Logger
func NewWithOptions(LogOptions) *Logger
```

Reply:
Added `New(io.Writer)` and `NewWithOptions(io.Writer, Options)`